airflow/providers/amazon/aws/executors/ecs/ecs_executor.py

```
@@ -288,6 +333,9 @@ def attempt_task_runs(self):
             _failure_reasons = []
             try:
                 run_task_response = self._run_task(task_key, cmd, queue, exec_config)
+            except (ClientError, NoCredentialsError):
```

Why are we raising on any ClientError here? Aren't there some ClientErrors that should follow the failure handling below?

---

That's a good catch. There needs to be a check here to make sure that the error is due to a credential issue. If so, we can raise the exception and have it get caught in the `sync` call.